gp-easy-passthrough/gpep-edit-entry.php

```diff
@@ -14,26 +14,98 @@ class GPEP_Edit_Entry {
+		$this->delete_partial = rgar( $options, 'delete_partial', true );
+
+		add_filter( "gform_entry_id_pre_save_lead_{$this->form_id}", array( $this, 'update_entry_id' ), 10, 2 );
+		add_filter( "gpls_rule_groups_{$this->form_id}", array( $this, 'bypass_limit_submissions' ), 10, 2 );
+
 	}
 
 	public function update_entry_id( $entry_id, $form ) {
-		if ( is_callable( 'gp_easy_passthrough' ) ) {
-			$session         = gp_easy_passthrough()->session_manager();
-			$update_entry_id = $session[ gp_easy_passthrough()->get_slug() . '_' . $form['id'] ];
-			if ( $update_entry_id && $this->delete_partial ) {
+
+		$update_entry_id = $this->get_update_entry_id( $form['id'] );
+		if ( $update_entry_id ) {
+			if ( $this->delete_partial
+			     && is_callable( array( 'GF_Partial_Entries', 'get_instance' ) )
+			     && $entry_id !== null
+			     && ! empty( GF_Partial_Entries::get_instance()->get_active_feeds( $form['id'] ) )
+			) {
 				GFAPI::delete_entry( $entry_id );
 			}
+			return $update_entry_id;
+		}
+
+		return $entry_id;
+	}
+
+	public function bypass_limit_submissions( $rule_groups, $form_id ) {
+
+		// Bypass GPLS if we're updating an entry.
+		if ( $this->get_update_entry_id( $form_id ) ) {
+			$rule_groups = array();
+		}
+
+		return $rule_groups;
+	}
+
+	public function get_update_entry_id( $target_form_id ) {
+
+		if ( ! is_callable( 'gp_easy_passthrough' ) ) {
+			return false;
+		}
+
+		$feeds = gp_easy_passthrough()->get_active_feeds( $target_form_id );
+		if ( empty( $feeds ) ) {
+			return false;
+		}
+
+		$session         = gp_easy_passthrough()->session_manager();
+		$update_entry_id = false;
+
+		foreach ( $feeds as $feed ) {
+			// @todo evaluate conditional logic before processing feed...
+			$source_form_id = $feed['meta']['sourceForm'];
+			// GPEP doesn't check for the user's last entry until the form is being rendered... but GPLS needs to know
+			// if we're updating an entry during validation which happens beforehand. Let's do the same legwork GPEP
+			// will do later to check for the last submitted entry when that option is active.
+			if ( rgars( $feed, 'meta/userPassthrough' ) && is_user_logged_in() ) {
+				$last_submitted_entry = GFAPI::get_entries(
+					$source_form_id,
+					array(
+						'field_filters' => array(
+							array(
+								'key'   => 'created_by',
+								'value' => get_current_user_id(),
+							),
+						),
+						'status'        => 'active',
+					),
+					array(
+						'key'       => 'date_created',
+						'direction' => 'DESC',
+					),
+					array( 'page_size' => 1 )
+				);
+				$update_entry_id      = rgars( $last_submitted_entry, '0/id', false );
+			}
+		}
+
+		$has_token           = ! empty( rgget( 'ep_token' ) );
+		$no_token_diff_forms = ! $has_token && (int) $target_form_id !== (int) $source_form_id;
+
+		if ( ! $update_entry_id && ( $has_token || $no_token_diff_forms ) ) {
+			$update_entry_id = $session[ gp_easy_passthrough()->get_slug() . '_' . $target_form_id ];
```

Comment:
`rgar` here to avoid potential PHP notice?

Reply:
@claygriffiths Fixed in [this commit](https://github.com/gravitywiz/snippet-library/pull/121/commits/822095a1a979e2f071ca18f0d070eaed82d65986)